gpiozero/output_devices.py

```diff
@@ -535,109 +536,123 @@ class TonalBuzzer(SourceMixin, CompositeDevice):
-        self._mid_note = self._note_to_midi(mid_note)
-        self._octaves = octaves
-        if octaves < 1:
-            raise ValueError('octaves must be at least 1')
-        if self.min_note < 1:
-            plural = 's' if octaves > 1 else ''
-            raise ValueError(
-                'mid_note too low for {} octave{}'.format(octaves, plural))
-
-    def note_value(self, note):
-        """
-        Convert the given note to a normalized value scaled -1 to 1 relative to
-        the buzzer's range. If note is ``None``, ``None`` is returned.
-        """
-        if note is None:
-            return None
-        note = self._note_to_midi(note)
-        return (note - self.mid_note) / (self.max_note - self.mid_note)
+            return super(TonalBuzzer, self).__repr__()
 
-    def play(self, note):
+    def play(self, tone):
         """
-        Play the given note e.g. ``play(60)`` (MIDI) or ``play('C4')`` (note).
-        ``play(None)`` turns the buzzer off.
+        Play the given *tone*. This can either be an instance of
+        :class:`~gpiozero.tones.Tone` or can be anything that could be used to
+        construct an instance of :class:`~gpiozero.tones.Tone`.
+
+        For example::
+
+            >>> from gpiozero import TonalBuzzer
+            >>> from gpiozero.tones import Tone
+            >>> b = TonalBuzzer(17)
+            >>> b.play(Tone("A4"))
+            >>> b.play(Tone(220.0)) # Hz
+            >>> b.play(Tone(60)) # middle C in MIDI notation
+            >>> b.play("A4")
+            >>> b.play(220.0)
+            >>> b.play(60)
         """
-        if note is None:
+        if tone is None:
             self.value = None
         else:
-            freq = self._note_to_freq(note)
-            if self.min_frequency <= freq <= self.max_frequency:
+            if not isinstance(tone, Tone):
+                tone = Tone(tone)
+            freq = tone.frequency
+            if self.min_tone.frequency <= tone <= self.max_tone.frequency:
                 self.pwm_device.pin.frequency = freq
                 self.pwm_device.value = 0.5
             else:
-                raise ValueError("note out of the device's range")
+                raise ValueError("tone is out of the device's range")
 
     def stop(self):
         """
         Turn the buzzer off. This is equivalent to setting :attr:`value` to
-        ``None``.
+        :data:`None`.
         """
         self.value = None
 
+    @property
+    def tone(self):
+        """
+        Returns the :class:`~gpiozero.tones.Tone` that the buzzer is currently
+        playing, or :data:`None` if the buzzer is silent. This property can
+        also be set to play the specified tone.
+        """
+        if self.pwm_device.pin.frequency is None:
+            return None
+        else:
+            return Tone.from_frequency(self.pwm_device.pin.frequency)
+
+    @tone.setter
+    def tone(self, value):
+        self.play(value)
```

Comment:
I'm not sure if it'd be more consistent with the rest of the GpioZero API to simply get rid of the `play()` method and just have a `tone` property-setter?

Reply:
I partly left `play()` in because it was in Claire's original design and I didn't want to alter it too much, and also because a fair number of tests depended on it and I'm lazy :). Finally I think `b.play("A4")` is a little more obvious to new coders than `b.tone = Tone("A4")` which is a bit less imperative (not that that's a bad thing, but it's nice to have the choice for newbies).

That said, I agree entirely that I shouldn't be messing with the PWM device's frequency (directly) in two different places. Given `play()` is staying, its setter should probably stay as a simple alias of `play()`, but either `play()` should set `value`, or `value` setter should call `play()`. I'll have a look at what leads to the simpler code (might be the latter as there'll be less messing with logs).

Reply:
Oh, could `b.tone = "A4"` not be made to work?
The only reason I brought this up is because e.g. https://gpiozero.readthedocs.io/en/master/api_output.html#angularservo only has an `angle` property and not a `move_to(angle)` method :wink: 
And it's obviously easier to add methods to an API later, than it is to add them now and remove them later. :man_shrugging:

Reply:
Hmm, I like it - but to make it consistent really ought to accept anything that could be used to construct a `Tone` which leads back to the ambiguity question below re: ints in the range 0..127 (which Ben's suggestion of keyword args may mitigate with the explicit `Tone` constructor).

 I could raise a `Warning` in the constructor if an int in the ambiguous range (1..127) was used but it's worth bearing in mind that would trigger for pretty much *any* MIDI note, if not explicitly constructed.

Reply:
Would it make sense to expose the various Tone constructors as property-setters in the TonalBuzzer? i.e. have `b.note = "A4"` be equivalent to `b.tone = Tone(note="A4")` (or `b.tone = Tone.from_note("A4")`), have `b.midi = 60` be equivalent to `b.tone = Tone(midi=60)` etc. Or does that become too wasteful of temporary objects?

LOL, maybe there needs to be a `TonesSetterMixin` :rofl: